src/Mvc/Mvc.Core/src/Infrastructure/ControllerActionInvokerCache.cs

```diff
@@ -43,30 +36,16 @@ internal class ControllerActionInvokerCache
             _mvcOptions = mvcOptions.Value;
         }
 
-        private InnerCache CurrentCache
-        {
-            get
-            {
-                var current = _currentCache;
-                var actionDescriptors = _collectionProvider.ActionDescriptors;
-
-                if (current == null || current.Version != actionDescriptors.Version)
```

Comment:
Don't we still need the IActionDescriptorCollectionProvider so we can the current ActionDescriptor.Version matches ActionDescriptorCollection.Version and invalidate the CacheEntry?